Feature/dx 273 ipfs download

W3stside opened the pull request:
# From Token uploading > Token list IPFS downloading

1. `TokenUpload` changed to work with new logic. Needs a name change probably.
2. Removed unncessarry and bulky state + fns that are no longer being used
3. changed styling to fit new `input` `type=text`

For a test file to grab: `QmZjVUBmJCnEytBycP8dciDbKCQje7tcZXvHn1RjsqcuDd` (may be gone)

To add file via `CLI`:
1. Tab1: `jsipfs daemon`
_Navigate to folder containing `tokens.json`_
2. Tab2: `jsipfs add tokens.json` (will return fileHash if successful)
_To check if u can cat file_
(optional) `jsipfs cat fileHashFromStep2`

What needs to change:
- [x] Logic if incorrect HASH is input in text and download is clicked - as of now nothing happens - `async` logic does NOT throw via `IPFSHOC` in `handleGrabFromIPFS` in function `const fileContent = await ipfsGetAndDecode(fileHash)` <- this should do something but it doesnt.
- [x] Bug checking around logic
- [x] Optimization

Have a good month bro! If you need anything, just let me know, I am on holiday but I'm not a dick and don't want you to feel stressed/shitty/helpless my man. Just message me if u need anything and I'll try to get back to you asap.

W3stside committed:
created setIPFSHash action and reducer

W3stside committed:
changed setNewIPFSCustomListAndUpdateBalances to not set hash and path

W3stside committed:
cleaned up IPFSHOC to use only necessary functions

1. removed handleSend, handleFileUpload
2. moved logic into handleGrabFromIPFS

W3stside committed:
removed unecessary state and dispatches
1. removed unused data and fns
2. created crude hash checker based on IPFS hash length, needs to change

W3stside committed:
changed component to use input type text and relevant new props
1. switched logic to work with hash as text instead of file upload

W3stside committed:
new styles for input text and surrounding components

W3stside committed:
Added internal State//new Logic//timeout to IPFSHOC

1. internal State to pass async data prop
2. added timeout logic in race if nothing pulled back in set time
3. better error handling

W3stside committed:
Changed logic / presentation of TokenUpload

1. Using Loader to conditionally render TokenUpload based on HOC internal state
2. changed wording

W3stside committed:
styling for Loader message//buttons//form//input

1. Loader message now bolded
2. buttonsCTA now smaller font and use min-height over padding
3. form 5% wider
4. ipfsDownload button has min-width
5. input now wide enough to git 46 character IPFS hash

W3stside committed:
reSized Loader to match prev Component size

W3stside committed:
clear input when incorrect hash entered + style changes

Velenir commented on src/actions/tokenList.ts:
```diff
@@ -12,11 +12,8 @@ export const setCustomTokenList = createAction<{ customTokenList: DefaultTokenOb
-    setCustomTokenList(p1),
-    setIPFSFileHashAndPath(p2),
-  ], 'BATCH_SET_CUSTOM_LIST_AND_IPFS_HASH_PATH'))
+export const setNewIPFSCustomListAndUpdateBalances = (p1: { customTokenList: DefaultTokenObject[] }) => async (dispatch: Function, getState: () => State) => {
```
does it need to be a thunk? it doesn't use state or have any async logic

W3stside replied:
what do u mean? it uses `customTokenList` and calculates `tokenBalances` `async`hronously

Velenir replied:
Ignore me, you're right. I was looking at some other action

Velenir commented on src/containers/TokenUpload/index.ts:
```diff
@@ -9,32 +14,26 @@ import { State } from 'types'
-  fileContent,
   json,
   customTokenList,
+  potentiallyValidHash: fileHash.length == 46,
```
`===`?

Velenir commented on src/components/IPFSHOC/index.tsx:
```diff
@@ -1,128 +1,77 @@
         console.error(error)
+        // reset hash to empty string
+        setIPFSFileHash('')
+
```
you never change state on caught error. Like `this.setState({ pullingData: false, error })`

Velenir commented on src/containers/TokenUpload/index.ts:
```diff
@@ -9,32 +14,26 @@ import { State } from 'types'
     setTokenListType,
-    setUploadFileParams,
+    openModal,
   })(IPFSHOC(TokenUpload) as any)
```
`TokenUpload` is wrapped in `IPFSHOC` twice, here and in `TokenUpload` component export

Velenir replied:
Still getting timeouts on seemingly valid hashes.

Velenir replied:
We may need to account for `hash+path` in input
like `QmYwAPJzv5CZsnA625s3Xf2nemtYgPpHdWEz79ojWnPbdG/readme`, leading to https://gateway.ipfs.io/ipfs/QmYwAPJzv5CZsnA625s3Xf2nemtYgPpHdWEz79ojWnPbdG/readme

W3stside replied:
wow good catch wtf was I thinking?

W3stside replied:
good point about the `path`ing but I am not really sure rn what do think on that

W3stside committed:
Fixed changes as requested in PR

1. set internal state back to default in IPFSHOC when error in pulling file
2. Unwrapped TokenUpload component from IPFSHOC as already wrapping in container/tokenUpload
3. == > === in length check
